### PR TITLE
SSP 4.99.148

### PR DESCRIPTION
**Pull request by cPanelChrisD**

### CODE REVIEW

File: ssp [hunk 1]
```diff
@@ -1913,28 +1911,20 @@ sub check_for_clustering {
...
-            }
-            else {
-                $cluster_member_hostname = '?';
-            }
-
-            if ( !$cluster_member_hostname ) {
-                $cluster_member_hostname = '?';
             }
-            if ( ${cluster_member} =~ m{ \A (vps\.net|softlayer) \z}xmsi ) {
+            $cluster_member_hostname = defined $cluster_conf{host} ? $cluster_conf{host} : '?';
```
Comment (davelcpanelnet-zz) [inline ssp:1926]: Curious of this ternary.  
What is the value of '?' at this point?

File: ssp [hunk 2]
```diff
@@ -1944,22 +1934,17 @@ sub check_for_clustering {
                 }
                 close $file_fh;
             }
-            else {
-                $cluster_member_role = '?';
-            }
+            $cluster_member_role = defined $cluster_member_role ? $cluster_member_role : '?';
```
Comment (davelcpanelnet) [inline ssp]: Ternary again here.  Going to look at this in debug

**Comment by davelcpanelnet-zz** [comment]
I'd like to see the ordering a abit different but that's a new TECH case I can file

[INFO] * DNS Clustering: is enabled
[INFO] * Apache: [ EA4 ] [ 2.4.23 ] [ Aug 24 2016 19:30:51 w/ ea-apache24-2.4.23-3.3.1.cpanel.x86_64 ]
[INFO] * Apache: is listening on ports [ 443 80 ]

Apache data should be first and foremost.

File: ssp (hunk 1 above)
Comment (davelcpanelnet-zz) [reply ssp:1926]: I got it
```
[INFO] * DNS Clustering: is enabled
	 \_ d33cec79-a0e1-4af8-9e7c-73c50eea0311.test.cpanel.net 10.17.59.128 [sync]
```

Looks good

File: ssp [hunk 3]
```diff
@@ -3137,6 +3122,92 @@ sub check_for_sneaky_htaccess {
...
 }
 
+sub check_ea4_paths_conf {
+    return unless $EA4;
+    my $paths_conf = '/etc/cpanel/ea4/paths.conf';
+    lstat($paths_conf);    # Can now use _ for file tests.
+
+    if ( !-e _ ) {
+        print_warn('EA4: ');
+        print_warning('/etc/cpanel/ea4/paths.conf is missing!');
```
Comment (davelcpanelnet-zz) [inline ssp:3132]:
```
[WARN] * EA4: /etc/cpanel/ea4/paths.conf is missing!
```

Looks good on the detection and display

File: ssp [hunk 4]
```diff
@@ -3137,6 +3122,92 @@ sub check_for_sneaky_htaccess {
...
+        return;
+    }
+    if ( !-f _ ) {
+        print_warn('EA4: ');
+        print_warning('/etc/cpanel/ea4/paths.conf is not a normal file!');
+        return;
+    }
+    if ( -z _ ) {
+        print_warn('EA4: ');
+        print_warning('/etc/cpanel/ea4/paths.conf is empty!');
```
Comment (davelcpanelnet-zz) [inline ssp:3142]:
```
[WARN] * EA4: /etc/cpanel/ea4/paths.conf is empty!
```

File: ssp [hunk 5]
```diff
@@ -3137,6 +3122,92 @@ sub check_for_sneaky_htaccess {
...
+        return;
+    }
+    if ( !-T _ ) {
+        print_warn('EA4: ');
+        print_warning('/etc/cpanel/ea4/paths.conf does not appear to be an ASCII text file!');
```
Comment (davelcpanelnet-zz) [inline ssp:3147]:
```
[WARN] * EA4: /etc/cpanel/ea4/paths.conf does not appear to be an ASCII text file!
```

File: ssp [hunk 6]
```diff
@@ -3137,6 +3122,92 @@ sub check_for_sneaky_htaccess {
...
+            return;
+        }
+        foreach my $key ( sort keys %conf ) {
+            if ( !exists $default_conf{$key} ) {                   # EA4 appears to ignore any unknown options, but count them.
+                $unknown_count++;
+                next;
+            }
+            if ( $default_conf{$key} ne $conf{$key} ) {
+                print_warn('EA4: ');
+                print_warning( '/etc/cpanel/ea4/paths.conf non-default setting: [ ' . $key . '=' . $conf{$key} . ' ]' );
```
Comment (davelcpanelnet-zz) [inline ssp:3196]:
```
[WARN] * EA4: /etc/cpanel/ea4/paths.conf non-default setting: [ dir_docroot=/usr/local/cpanel/t/qa/www/ ]
[WARN] * EA4: /etc/cpanel/ea4/paths.conf non-default setting: [ dir_run=/etc/apache2/run ]
```

File: ssp [hunk 7]
```diff
@@ -8864,12 +8950,6 @@ sub check_ea4_php_fpm {
     return 1;
 }
 
-sub check_for_missing_fstab {
-    return if -e "/etc/fstab";
-    print_warn('Filesystem: ');
-    print_warning("/etc/fstab may be missing. Can cause /scripts/fixquotas to fail. See case CPANEL-6082.");
-}
-
```
Comment (davelcpanelnet-zz) [inline ssp:8872]: Seems covered on
```
3291        '/etc/fstab'                            => { mode => ['0644'], check_missing => 1, perms_help => 'Missing fstab can break /scripts/fixquotas (CPANEL-6082), and bad perms can break cPanel U\
    I (CPANEL-11021).' },
```

**Pull request merged**